src/utils.ts

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2017 MTRNord and Cooperative EITA
+Copyright 2017 Vector Creations Ltd.
+Copyright 2019 Michael Telatynski <7t3chguy@gmail.com>
+Copyright 2019 - 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Returns a language string with underscores replaced with
+ * hyphens, and lowercased.
+ *
+ * @param {string} language The language string to be normalized
+ * @returns {string} The normalized language string
+ */
+export function normalizeLanguageKey(language: string): string {
+    return language.toLowerCase().replace("_", "-");
+}
+
+/**
+ * Turns a language string, normalises it,
+ * (see normalizeLanguageKey) into an array of language strings
+ * with fallback to generic languages
+ * (eg. 'pt-BR' => ['pt-br', 'pt'])
+ *
+ * @param {string} language The input language string
+ * @return {string[]} List of normalised languages
```

Comment:
Do types still belong in doc comments like this, now that everything is TypeScript?